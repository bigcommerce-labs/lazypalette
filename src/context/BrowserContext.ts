import React from 'react';

export interface Browser {
    _window: any;
}

const defaultValue: Browser = {
    _window: window,
};

const BrowserContext = React.createContext(defaultValue);

export default BrowserContext;
