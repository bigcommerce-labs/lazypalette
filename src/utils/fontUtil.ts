export function parseFont(font: string) {
    const provider = font.split('_')[0];
    switch (provider) {
        case 'Google':
            return parseGoogleFont(font);
        default:
            return null;
    }
}

function parseGoogleFont(font: string) {
    const split = font.split('_');
    const family = split[1];
    let formattedFont = '';
    let weight = split[2];

    if (split.length === 2) {
        formattedFont += `${family}|`;
    } else if (split.length > 2) {
        weight = weight.split(',')[0];
        formattedFont += `${family}:${weight}|`;
    }

    return `//fonts.googleapis.com/css?family=${formattedFont}`;
}
