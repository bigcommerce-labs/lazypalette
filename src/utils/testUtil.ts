export const sel = (id: string) => `[data-test-id="${id}"]`;
