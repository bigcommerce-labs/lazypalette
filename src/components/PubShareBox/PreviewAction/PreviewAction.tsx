import React, {SFC} from 'react';

import BrowserContext, {Browser} from '../../../context/BrowserContext';
import { purchaseEndPoint } from '../constants';

import ButtonInput from '../../ButtonInput/ButtonInput';
import { ButtonWrapper } from '../styles';

interface PreviewActionProps {
    price: number;
    variationId: string;
}

const PreviewAction: SFC<PreviewActionProps> = props => {
    const { price, variationId } = props;
    const themePrice = price > 0 ? (price / 10).toFixed(2) : price;
    const actionText = price > 0 ? `Buy $${themePrice}` : 'Add Theme';
    const actionTestId = price > 0 ? 'buy' : 'add-theme';

    return (
        <>
            <BrowserContext.Consumer>
                {({ _window }: Browser) =>
                    <ButtonWrapper>
                        <ButtonInput
                            classType="primary"
                            onClick={() => _window.location.assign(purchaseEndPoint(variationId))}
                            type="button"
                            testId={actionTestId}
                            disabled={false}
                        >
                            {actionText}
                        </ButtonInput>
                    </ButtonWrapper>
                }
            </BrowserContext.Consumer>
        </>
    );
};

export default PreviewAction;
