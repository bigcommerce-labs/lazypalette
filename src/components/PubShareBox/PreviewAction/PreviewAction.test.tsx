import { mount, shallow } from 'enzyme';
import React from 'react';

import BrowserContext from '../../../context/BrowserContext';
import { sel } from '../../../utils/testUtil';

import PreviewAction from './PreviewAction';

describe('InactiveAction', () => {

    const buyButton = sel('buy');
    const addThemeButton = sel('add-theme');

    const mockHref = jest.fn();
    const mockWindow = {
        location: {},
    };

    Object.defineProperty(mockWindow.location, 'href', {
        set: mockHref,
    });

    const PreviewActionElement = <PreviewAction
        price={100}
        variationId={'234'}
    />;

    const previewActionShallow = shallow(PreviewActionElement);

    it('renders', () => {
        expect(previewActionShallow).toMatchSnapshot();
    });

    describe('when price is zero', () => {
        const previewActionMount = mount(
            <BrowserContext.Provider value={{ _window: mockWindow }}>
                <PreviewAction
                    price={0}
                    variationId={'234'}
                />
            </BrowserContext.Provider>
        );
        it('should have "Add Theme" button', () => {
            previewActionMount.setProps({price: 0});
            expect(previewActionMount.find(addThemeButton).hostNodes().length).toBe(1);
        });
    });

    describe('when price is greater than 0', () => {
        const previewActionMount = mount(
            <BrowserContext.Provider value={{ _window: mockWindow }}>
                <PreviewAction
                    price={100}
                    variationId={'234'}
                />
            </BrowserContext.Provider>
        );
        it('should have a "Buy" button', () => {
            previewActionMount.setProps({price: 100});
            expect(previewActionMount.find(buyButton).hostNodes().length).toBe(1);
        });
        describe('when price is 100', () => {
            it('buy button has the text "Buy $10.00"', () => {
                previewActionMount.setProps({price: 100});
                expect(previewActionMount.find(buyButton).hostNodes().text()).toBe('Buy $10.00');
            });
        });
    });

    describe('when you click on Buy or Add Theme', () => {
        const previewActionMount = mount(
            <BrowserContext.Provider value={{ _window: mockWindow }}>
                <PreviewAction
                    price={100}
                    variationId={'234'}
                />
            </BrowserContext.Provider>
        );
        it('should redirect to /manage/marketplace/themes/234?action=purchase', () => {
            previewActionMount.setProps({price: 100});
            previewActionMount.find(buyButton).hostNodes().simulate('click');
            expect(mockHref).toBeCalled();
        });
    });
});
