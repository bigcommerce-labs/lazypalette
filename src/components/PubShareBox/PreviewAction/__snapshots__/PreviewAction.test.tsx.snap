// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InactiveAction renders 1`] = `
<Fragment>
  <ContextConsumer>
    <Component />
  </ContextConsumer>
</Fragment>
`;
